### PR TITLE
feat(depinject): support for ica

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

Adding support for ica package.

ref: https://github.com/cosmos/ibc-go/issues/3560

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/27-interchain-accounts/depinject.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+// ModuleInputs defines the 27-interchain-accounts module inputs for depinject.
+type ModuleInputs struct {
+	depinject.In
+
+	Config *modulev1.Module
+	Cdc    codec.Codec
+	Key    *storetypes.KVStoreKey
+
+	Ics4Wrapper   porttypes.ICS4Wrapper
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/depinject.go:41]: this is just a nit, dismiss if you want, hopefully we kill ics4wrapper at some point

```suggestion
	ICS4Wrapper   porttypes.ICS4Wrapper
```

File: modules/apps/27-interchain-accounts/depinject.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	)
+}
+
+// ModuleInputs defines the 27-interchain-accounts module inputs for depinject.
+type ModuleInputs struct {
+	depinject.In
+
+	Config *modulev1.Module
+	Cdc    codec.Codec
+	Key    *storetypes.KVStoreKey
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/depinject.go:39]: There is two different store keys for icacontroller and icahost, looks like this is passing one for both keepers.

See https://github.com/cosmos/ibc-go/blob/main/testing/simapp/app.go#L304

**Review by damiannolan — CHANGES_REQUESTED** [review]
Just requesting change on store keys, the rest should be good to go for now! 
Same point applies about app registration and `Invoker` for core.

Thanks @crodriguezvega!

File: modules/apps/27-interchain-accounts/depinject.go (hunk 2 above)
Comment (crodriguezvega) [reply modules/apps/27-interchain-accounts/depinject.go]: Good catch. I fixed it now.

**Pull request merged**